hw/ip/pattgen/doc/_index.md

```diff
@@ -0,0 +1,143 @@
+A one in the polarity bit inverts the `pcl` clock so that it starts high and `pda` transitions on the rising edge.
+{{<wavejson>}}
+{signal: [
+  {name: 'PATT_CTRL.ENABLE_CH0', wave: 'lh......'},
+  {name: 'PATT_CTRL.POLARITY_CH0 (default: low)', wave: '0.......'},
+  {name: 'pcl0_tx', wave: '0.hlhlhl'},
+  {name: 'pda0_tx', wave: 'x3.3.3.3', data: 'DATA[0] DATA[1] DATA[2]'},
+  {name: 'PATT_CTRL.POLARITY_CH1 (high)', wave: '1.......'},
+  {name: 'pcl1_tx', wave: '1.lhlhlh'},
+  {name: 'pda1_tx', wave: 'x5.5.5.5', data: 'DATA[0] DATA[1] DATA[2]'},
+  
+],
+  head: {text: 'Effect of the Polarity Registers',tick:0}}
+{{</wavejson>}}
+
+1. Program the seed patterns (1-64 bits in length) using the multi-registers {{< regref "PATT_DATA_CH0" >}} (for Channel 0) or {{<regref "PATT_CH1_DATA" >}} (for Channel 1).
+For either channel, the first 32 bits to be transmitted, are programmed in the lower half of the multi-register (e.g. {{< regref "PATT_DATA_CH0_0" >}}), and the latter 32 bits are programmed in the upper half of multi-register (e.g. {{< regref "PATT_DATA_CH0_1" >}}).
+1. Program the clock divider ratios to the appropriate registers, {{< regref "PATT_PREDIV_CH0.CLK_RATIO" >}} for Channel 0 or {{< regref "PATT_PREDIV_CH1.CLK_RATIO" >}} for Channel 1.
+The resulting clock frequency will be slower than the input I/O clock by a ratio of 2&times;(CLK_RATIO+1):
+$$f_{pclx}=\frac{f_\textrm{I/O clk}}{2(\textrm{CLK_RATIO}+1)}$$
+1. Program the length of seed pattern using one of the fields {{< regref "PATT_SIZE.LEN_CH0" >}}, for Channel 0. or {{< regref "PATT_SIZE.LEN_CH1" >}} for Channel 1.
+Note that since the allowed seed length ranges from 1-64, the value of this field should be one less than the pattern length.
+For example, to generate an 16-bit pattern on Channel 1, a value of 15 should be written to the field {{< regref "PATT_SIZE.LEN_CH1" >}}. 
+1. Program the desired number of pattern repetitions using one of the fields {{< regref "PATT_SIZE.REPS_CH0" >}} for Channel 0, or {{< regref "PATT_SIZE.REPS_CH1" >}} for Channel 1.
+Note that since the allowed number of pattern repetitions ranges from 1-1024, the value of this field should be one less than the desired repetition count.
+For example, to repeat a pattern 30 times on Channel 1, a value of 29 should written to the field {{< regref "PATT_SIZE.REPS_CH1" >}}. 
+1. Finally to start the pattern, set the {{< regref "PATT_CTRL.ENABLE_CH0" >}} to enable Channel 0 or set {{< regref "PATT_CTRL.ENABLE_CH1" >}} to enable Channel 1.
+Set both bits simultaneously, to start both channel patterns at the same time.
+
+The pattern can be halted on either channel by setting the corresponding `PATT_CTRL.ENABLE` bit to 0 for the desired channel.
+Note that re-enabling either channels restarts the pattern sequence for that channel from the beginning.
+
+PLease note that writes to a channel's configuration registers have no effect while a pattern is running on that channel.
```

Comment:
```suggestion
Please note that writes to a channel's configuration registers have no effect while a pattern is running on that channel.
```

Can you mention this constraint in the hjson in the description of all affected configuration registers? It's very easy to miss such a sentence in the docs and wonder why the changes are not applied.

Reply:
Ah.  Sure.  Nice suggestion.